examples/nodes/utils.py

```diff
@@ -0,0 +1,300 @@
+        return out
+
+
+def train_mnist_model(model, num_epochs, train_loader, test_loader, loss_fn, optimizer):
+    """
+    Trains a PyTorch model on the MNIST dataset.
+    Args:
+        model (nn.Module): The PyTorch model to be trained.
+        num_epochs (int): The number of epochs to train the model for.
+        train_loader (nodes.Loader): A DataLoader object containing the training data.
+        test_loader (nodes.Loader): A DataLoader object containing the testing data.
+        loss_fn (nn.Module): The loss function to be used during training.
+        optimizer (Optimizer): The optimizer to be used during training.
+    Returns:
+        tuple: A tuple containing the trained model, a list of losses at each epoch, and a list of accuracies at each epoch.
+    Notes:
+        This function trains the model using the provided training data and evaluates its performance on the provided testing data.
+    """
+    loss_list = []
+    accuracy_list = []
+    for epoch in range(num_epochs):
+        num_loops = 0
+        total_loss = 0
+        for batch in train_loader:
+            images = batch["image"]
+            labels = batch["label"]
+            outputs = model(images)
+            loss = loss_fn(outputs, labels)
+            # Backward and optimize
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+            num_loops += 1
+        total_loss /= num_loops
+        loss_list.append(total_loss)
+
+        # Test the model
+        with torch.no_grad():
+            n_correct = 0
+            n_samples = 0
+            for batch in test_loader:
+                images = batch["image"]
+                labels = batch["label"]
+                outputs = model(images)
+                _, predicted = torch.max(outputs.data, 1)
+                n_samples += labels.size(0)
+                n_correct += (predicted == labels).sum().item()
+
+            acc = 100.0 * n_correct / n_samples
+            accuracy_list.append(acc)
+            if epoch == num_epochs - 1:
+                print(f"Test Accuracy: {acc} %")
+    return model, loss_list, accuracy_list
+
+
+def csv_reader(filename):
+    """
+    Reads a CSV file and returns its contents as a list of lists.
+    Args:
+        filename (str): The path to the CSV file to be read.
+    Returns:
+        list: A list of lists, where each inner list represents a row in the CSV file.
+    Notes:
+        This function assumes that the CSV file has a header row that should be skipped.
+        It also assumes that each row in the CSV file contains only numeric values.
+    """
+    with open(filename) as csvfile:
+        data = []
+        for line_number, line in enumerate(csvfile):
+            if line_number == 0:
+                continue
+            row = [float(x) for x in line.strip().split(",")]
+            data.append(row)
+    return data
+
+
+class MDS_Net(nn.Module):
```

Comment:
The model like a simple MLP. What does MDS stand for? multidataset?

Reply:
Yes, but if it is confusing I can modify the name to something simpler.

Reply:
Just call this MLP